tests/Date/ConstructTest.php

```diff
@@ -258,5 +249,110 @@ public function testConstructWithRelative($class)
     {
         $c = new $class('+7 days');
         $this->assertEquals('00:00:00', $c->format('H:i:s'));
+
+        $c = new $class('+10 minutes');
+        $this->assertEquals('00:00:00', $c->format('H:i:s'));
+
+        $c = new $class('2001-01-01 +7 days');
+        $this->assertEquals('2001-01-08', $c->format('Y-m-d'));
+    }
+
+    /**
+     * @dataProvider dateClassProvider
+     */
+    public function testConstructWithLocalTimezone($class)
+    {
+        $londonTimezone = new DateTimeZone('Europe/London');
+
+        // now adjusted to London time
+        // This test could have different results depending on when now is
+        $c = new $class('now', $londonTimezone);
+        $london = new DateTimeImmutable('now', $londonTimezone);
+        $this->assertEquals($london->format('Y-m-d'), $c->format('Y-m-d'));
+        $this->assertSame('UTC', $c->tzName);
+
+        // Confirm that timezone is adjusted before midnight is set.
+        // This implies that the relative time used by Date helpers
+        // is affected by the timezone parameter properly.
+        $london = new DateTimeImmutable('midnight', $londonTimezone);
+        $this->assertEquals('00:00:00', $london->format('H:i:s'));
```

Comment:
When would this fail? It looks like it is testing PHP.

Reply:
This is me being paranoid about testing.  This *is* testing PHP, but it's testing the premise behind Date using the timezone parameter.

I wanted to find a way to test Date with 'now' or 'today' without using TestNow.

Reply:
Oh, when would it fail?  It would fail if PHP was doing: now > midnight > timezone.  That would mean local time is adjusted to 00:00:00 and then adjusted to the target timezone. 

This confirms it is doing: now > timezone > midnight.

Reply:
I'm not sure we need to be paranoid about how PHP is working, unless we're using it as a frame of reference to compare our results.

Reply:
I can remove it.  It was placeholder while I was testing everything else.